javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,243 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj){
+  // Solution code here...
+  Object.entries(obj).forEach(entry => {
+    console.log(entry);
+    console.log(entry[0]);
+    let key = entry[0];
+    let value = entry[1];
+    const li = document.createElement('li');
+    li.textContent = `${key}: ${value}`;
+    return li;
+  });
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  // Solution code here...
+  let count = 0;
+  input.map(array => {
+    array.map(number => {
+      if (number === target) {
+        count++;
+      }
+    });
+  });
+  return count;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that, given an array of integer arrays as input, calculates the total sum of all the elements in the array.
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  // Solution code here...
+  console.log(input);
+  let totalReduced = input.reduce((a, b) => {
+    console.log('a', a);
+    console.log('b', b);
+    let aReduced = a.reduce((acc, current) => {
+      return acc+current;
+    });
+    console.log('aReduced', aReduced);
+    let bReduced = b.reduce((acc, current) => {
+      return acc+current;
+    });
+    console.log('bReduced', bReduced);
+    a = aReduced + bReduced;
+  });
+  return totalReduced;
```

Comment:
Good work so far on the challenges Beau!

We may be able to simplify this a bit by using reduce() once. Perhaps you can try using [flat()](https://dev.to/stlnick/javascript-array-flat-4l8f) first? 👍